go/plugins/ollama/embed.go

```diff
@@ -0,0 +1,152 @@
+}
+
+func embed(ctx context.Context, serverAddress string, req *ai.EmbedRequest) (*ai.EmbedResponse, error) {
+	options, ok := req.Options.(*EmbedOptions)
+	if !ok && req.Options != nil {
+		return nil, fmt.Errorf("invalid options type: expected *EmbedOptions")
+	}
+	if options == nil || options.Model == "" {
+		return nil, fmt.Errorf("invalid embedding model: model must be specified")
+	}
+
+	if serverAddress == "" {
+		return nil, fmt.Errorf("invalid server address: address cannot be empty")
+	}
+
+	ollamaReq := newOllamaEmbedRequest(options.Model, req.Documents)
+
+	jsonData, err := json.Marshal(ollamaReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal embed request: %w", err)
+	}
+
+	resp, err := sendEmbedRequest(ctx, serverAddress, jsonData)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("ollama embed request failed with status code %d", resp.StatusCode)
+	}
+
+	var ollamaResp ollamaEmbedResponse
+	if err := json.NewDecoder(resp.Body).Decode(&ollamaResp); err != nil {
+		return nil, fmt.Errorf("failed to decode embed response: %w", err)
+	}
+
+	return newEmbedResponse(ollamaResp.Embeddings), nil
+}
+
+func sendEmbedRequest(ctx context.Context, serverAddress string, jsonData []byte) (*http.Response, error) {
+	client := &http.Client{}
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", serverAddress+"/api/embed", bytes.NewBuffer(jsonData))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+	httpReq.Header.Set("Content-Type", "application/json")
+	return client.Do(httpReq)
+}
+
+func newOllamaEmbedRequest(model string, documents []*ai.Document) ollamaEmbedRequest {
+	var input interface{}
+	if len(documents) == 1 {
+		input = concatenateText(documents[0])
+	} else {
+		texts := make([]string, len(documents))
+		for i, doc := range documents {
+			texts[i] = concatenateText(doc)
+		}
+		input = texts
+	}
+
+	return ollamaEmbedRequest{
+		Model: model,
+		Input: input,
+	}
+}
+
+func newEmbedResponse(embeddings [][]float32) *ai.EmbedResponse {
+	resp := &ai.EmbedResponse{
+		Embeddings: make([]*ai.DocumentEmbedding, len(embeddings)),
+	}
+	for i, embedding := range embeddings {
+		resp.Embeddings[i] = &ai.DocumentEmbedding{Embedding: embedding}
+	}
+	return resp
+}
+
+func concatenateText(doc *ai.Document) string {
+	var builder strings.Builder
+	for _, part := range doc.Content {
+		builder.WriteString(part.Text)
+	}
+	result := builder.String()
+	return result
+}
+
+// DefineEmbedder defines an embedder with a given server address.
+func DefineEmbedder(serverAddress string, model string) ai.Embedder {
+	state.mu.Lock()
+	defer state.mu.Unlock()
+	if !state.initted {
+		panic("ollama.Init not called")
+	}
+	return ai.DefineEmbedder(provider, serverAddress, func(ctx context.Context, req *ai.EmbedRequest) (*ai.EmbedResponse, error) {
+		if req.Options == nil {
```

Comment:
does this also need 
```go
if req.Options.Model == "" {
  req.Options.Model = model
}
```
?

Otherwise it looks like `req.Options.Model` is required field